src/main/java/marquez/common/models/DatasourceUrn.java

```diff
@@ -33,8 +35,19 @@ public static DatasourceUrn from(@NonNull DatasourceType type, @NonNull Datasour
     return fromString(value);
   }
 
+  public static DatasourceUrn from(
+      @NonNull ConnectionUrl connectionUrl, @NonNull DatasourceName name) {
+    return from(connectionUrl.getDatasourceType(), name);
+  }
+
+  public static DatasourceUrn from(@NonNull String connectionUrl, @NonNull String name) {
+    return from(
+        ConnectionUrl.fromString(connectionUrl).getDatasourceType(),
+        DatasourceName.fromString(name));
+  }
```

Comment:
Do we need:

```
DatasourceUrn.from(ConnectionUrl, DatasourceName)
DatasourceUrn.from(String, String)
```

We only want to define `DatasourceUrn.from(DatasourceType, DatasourceName)`. Do the new builder methods add convenience when constructing datasource URNs?

Reply:
They are pretty convenient and make the code much less clunky, especially for test code. The code to extract the type from the `connectionUrl` can get verbose and muddy up the code.

I think it's also nice because it takes in the two attributes that a user gives to create a Datasource, namely the connection url and the name.

Reply:
Makes sense. We'll only want to define `DatasourceUrn.from(ConnectionUrl, DatasourceName)` and an provide an accompanying test for the new method.